plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java

```diff
@@ -186,35 +193,46 @@ public Content get(final MavenPath path) throws IOException {
   @Override
   public Content get(final StorageTx tx, final MavenPath path) throws IOException {
     log.debug("GET {} : {}", getRepository().getName(), path.getPath());
-    final Asset asset = findAsset(tx, tx.getBucket(), path);
-    if (asset == null) {
-      return null;
-    }
-    return toContent(tx, asset);
+    return toContent(tx, path);
   }
 
   /**
    * Creates {@link Content} from passed in {@link Asset}.
    */
-  private Content toContent(final StorageTx tx, final Asset asset) throws IOException {
-    final Blob blob = tx.requireBlob(asset.requireBlobRef());
-    final String contentType = asset.contentType();
-
-    final NestedAttributesMap checksumAttributes = asset.attributes().child(StorageFacet.P_CHECKSUM);
-    final Map<HashAlgorithm, HashCode> hashCodes = Maps.newHashMap();
-    for (HashAlgorithm algorithm : HashType.ALGORITHMS) {
-      final HashCode hashCode = HashCode.fromString(checksumAttributes.require(algorithm.name(), String.class));
-      hashCodes.put(algorithm, hashCode);
+  private Content toContent(final StorageTx tx, final MavenPath mavenPath) throws IOException {
+    IllegalStateException lastIllegalStateException = null;
```

Comment:
Be good to refactor apart the two functions this method has - the outer retrying logic, and the inner transformation logic.